Mage/src/main/java/mage/game/GameState.java

```diff
@@ -823,21 +823,19 @@ public void addSimultaneousDamage(DamagedEvent damagedEvent, Game game) {
         boolean isPlayerBatchUsed = false;
         for (GameEvent event : simultaneousEvents) {
 
-            // per damage type
-            if ((event instanceof DamagedBatchEvent)
-                    && ((DamagedBatchEvent) event).getDamageClazz().isInstance(damagedEvent)) {
-                ((DamagedBatchEvent) event).addEvent(damagedEvent);
-                isDamageBatchUsed = true;
-            }
-
-            // per player
             if (isPlayerDamage && event instanceof DamagedBatchForOnePlayerEvent) {
+                // per player
                 DamagedBatchForOnePlayerEvent oldPlayerBatch = (DamagedBatchForOnePlayerEvent) event;
                 if (oldPlayerBatch.getDamageClazz().isInstance(damagedEvent)
                         && event.getPlayerId().equals(damagedEvent.getTargetId())) {
                     oldPlayerBatch.addEvent(damagedEvent);
                     isPlayerBatchUsed = true;
                 }
+            } else if ((event instanceof DamagedBatchEvent)
```

Comment:
Probably we should add more comments here to clarify that this covers `DamagedBatchForPermanentsEvent` and `DamagedBatchForPlayersEvent`